### PR TITLE
Abort errors early

### PR DESCRIPTION
**Pull request by JeromeDeLeon**
I want to create another option like `abortEarly` but decided to reuse `validateAllFieldCriteria`.
Here's my attempt to solve it iwth [codesandbox](https://codesandbox.io/embed/goofy-gates-s9sv3?fontsize=14&hidenavigation=1&theme=dark) 

@bluebill1049 let me know your thoughts :smiley:

### CODE REVIEW

**Comment by bluebill1049** [comment]
looks good, maybe we can add unit tests for this as well. happy to merge it in tho. I can supply unit tests later on.

**Review by bluebill1049 — APPROVED** [review]

File: src/useForm.ts [hunk 1]
```diff
@@ -1014,10 +1014,9 @@ export function useForm<FormValues extends FieldValues = FieldValues>({
     () => () => {
       isUnMount.current = true;
       fieldsRef.current &&
-        Object.values(
-          fieldsRef.current,
-        ).forEach((field: Field | undefined): void =>
-          removeEventListenerAndRef(field, true),
+        Object.values(fieldsRef.current).forEach(
```
Comment (bluebill1049) [inline src/useForm.ts]: weird why your prettier is not working for you. :( there is nothing changed in here...

Comment (JeromeDeLeon) [reply src/useForm.ts]: Yeah, it automatically lints. Maybe because of the Prettier extension for VSCode

Comment (bluebill1049) [reply src/useForm.ts]: all good :)

Comment (bluebill1049) [reply src/useForm.ts]: thanks a lot for the PR @JeromeDeLeon ❤️

**Comment by JeromeDeLeon** [comment]
> looks good, maybe we can add unit tests for this as well. happy to merge it in tho. I can supply unit tests later on.

Sorry, I'm still learning unit testing, can't do much. I'm learning by looking on how do you test since your expert on this one :smile:

**Comment by bluebill1049** [comment]
no worries @JeromeDeLeon i will sort it out 👍

File: src/useForm.ts (hunk 1 above)
Comment (JeromeDeLeon) [reply src/useForm.ts]: You're welcome :smile:

**Pull request merged**